tests/MethodCallTest.php

```diff
@@ -196,6 +196,60 @@ function printInt(int $int): void {
     public function providerValidCodeParse(): iterable
     {
         return [
+            'nullSafeCallNotNullMakesVarNotNull' => [
+                'code' => '<?php
+                    class Foo {
+                        public function check(): bool {
+                            return false;
+                        }
+                        public function do(): void {}
+                    }
+
+                    /** @var ?Foo */
+                    $foo = null;
+
+                    if ($foo?->check() === null) {
+                        /** @psalm-check-type-exact $foo = null */
```

Comment:
I would be curious to see the same test but with check that returns "?bool". Because then you can't infer anything from the nullsafe call

Reply:
Yep my exact thoughts, I still need to finish this PR